chainer/training/extensions/parameter_statistics.py

```diff
@@ -0,0 +1,233 @@
+import numpy
+
+import cupy
+
+from chainer import cuda
+from chainer import reporter
+from chainer import training
+from chainer.training import extension
+
+
+def _iterable(x):
+    if isinstance(x, (list, tuple)):
+        return x
+    return x,
+
+
+def _prefix_statistics(prefix, stats):
+    """Prefix all keys in a statistic dictionary."""
+    for key in list(stats.keys()):
+        stats['{}/{}'.format(prefix, key)] = stats.pop(key)
+    return stats
+
+
+def _statistic_key(link, param_names, attr_names):
+    """Generate a statistic dictionary key based on context."""
+    param_names = _iterable(param_names)
+    attr_names = _iterable(attr_names)
+
+    link_name = 'None' if not hasattr(link, 'name') else link.name
+    param_name = '-'.join(param_names)
+    attr_name = '-'.join(attr_names)
+
+    return '{}/{}/{}'.format(link_name, param_name, attr_name)
+
+
+def _flatten_link(link, param_names, attr_names):
+    """Flatten a link into an array."""
+    param_names = _iterable(param_names)
+    attr_names = _iterable(attr_names)
+
+    params = []
+    for param in link.params():
+        if param.name in param_names:
+            for attr_name in attr_names:
+                p = getattr(param, attr_name)
+                p = p.flatten()
+                params.append(p)
+
+    return link.xp.concatenate(params)
+
+
+def _statistics(x, functions):
+    """Compute statisticts for the given array.
+
+    Args:
+        x (array): Target array for which statistics are computed.
+        functions (iterable): Statistics to collect, mapping directly to NumPy
+            or CuPy functions.
+
+    Returns:
+        dict: Mappings from functions keys to statistic values.
+    """
+    stats = {}
+    for f in functions:
+        try:
+            stats[f] = getattr(x, f)()
+        except ValueError:
+            stats[f] = float('NaN')
+    return stats
+
+
+def _percentiles(x, sigmas):
+    """Compute percentiles for the given array.
+
+    Args:
+        x (array): Target array for which percentiles are computed.
+        sigmas (iterable): Percentile sigma values.
+
+    Returns:
+        array: List of percentiles. The list has the same length as the given
+            ``sigma``.
+    """
+    def _percentiles_cpu(_x):
+        try:
+            return numpy.percentile(_x, sigmas)
+        except IndexError:
+            return numpy.array((float('NaN'),) * 7)
```

Comment:
Thanks a lot, changed to len(sigmas). Also, the error is not an expected behavior. Added comments in the code.